app/shared/layouts/colored-layout/ColoredLayout.tsx

```diff
@@ -0,0 +1,33 @@
+import { Box, BoxProps } from '@mui/material';
+import React from 'react';
+
+import { styles } from '~/layouts/colored-layout/ColoredLayout.styles';
+import MainLayout, { MainLayoutProps } from '~/layouts/main-layout/MainLayout';
+
+export interface ColoredLayoutProps extends BoxProps {
+  color?: string;
+  mainLayoutProps?: Omit<MainLayoutProps, 'children'>;
+}
+
+const ColoredLayout: React.FC<ColoredLayoutProps> = ({
+  mainLayoutProps = {},
+  color = '#F2EEE8',
+  children,
+  sx,
+  ...props
+}) => {
+  const { sx: mainLayoutSx, ...otherProps } = mainLayoutProps;
+
+  return (
+    <Box sx={[styles.container(color), ...(Array.isArray(sx) ? sx : [sx])]} data-testid="colored-layout" {...props}>
+      <MainLayout
+        sx={[styles.childrenBox, ...(Array.isArray(mainLayoutSx) ? mainLayoutSx : [mainLayoutSx])]}
+        {...otherProps}
+      >
+        {children}
+      </MainLayout>
+    </Box>
+  );
+};
```

Comment:
I don't think we should expose any styles of the background color box aside from `backgroundColor`. This way we can simplify `ColoredLayout` and make its props extend `MainLayoutProps` with optional color as prop. So we get someting like this:

```suggestion
export interface ColoredLayoutProps extends MainLayoutProps {
    color?: string;
}

const ColoredLayout: React.FC<ColoredLayoutProps> = ({
    color = '#F2EEE8',
    children,
    sx,
    ...props
}) => (
    <Box sx={{ backgroundColor: color }} data-testid="colored-layout">
        <MainLayout
            sx={[styles.childrenBox, ...(Array.isArray(sx) ? sx : [sx])]}
            {...props}
        >
            {children}
        </MainLayout>
    </Box>
);
```

Reply:
I understand that this is not the best practice, but I wanted to provide maximum flexibility and code reuse. Also, the container also has its own specific styles in the ColoredLayout.styles.ts file. Your version looks better, but it is less flexible.

Reply:
Sorry, i missed the styles part. But we don't really need this much of the flexibility. It like to premature optimisation but with flexibility, I think for now its better to make more understandable interface, bacuase we have no idea what will happen in the future, and refactoring simple interface is much easier than a flexible one

Reply:
I think we need a third party opinion.

Reply:
Agree

Reply:
Let's use Roman approach(two separate independent layouts). As for me it is more flexible and will avoid using unexpected styles.

Reply:
Well there still would be 2 layouts, just  `ColoredLayout` will inherit from `MainLayout`. Background Box only reason to exist is to provide background color without being limited by width of main layout, we really don't need to modify its `sx`, everything could be done through `MainLayout` that requires style changes as all content will be inside. So my opinion still stands, we don't need to manipulate styles of background and simply let it exist with hard-coded styles

Reply:
Now I understand what Yuriy was trying to convey, and I agree with it. I edited the layout, according to his wishes, but left the option to change the styles in the wrappers, if it is necessary at some point. However, I did not make both layouts independent, because, in my opinion, this violates the DRY principles, because the styles for the `MainLayout` will be duplicated in both components.

Reply:
Yes, this was almost the thing i was trying to say, excepti argued that we wouldn't need styles adjustments for background box but i can get behind that, you've got my approval :+1: